### PR TITLE
Adding the option to insert a social profile link

### PR DESCRIPTION
**Pull request by OzzyCzech**
### Request type

- [x] Feature
- [ ] Fix
- [ ] Refactor
- [x] Tests
- [x] Documentation

### Summary

There are [several ways](https://alessandrorossini.org/the-sad-story-of-the-vcard-format-and-its-lack-of-interoperability/) to solve social networks profile on VCARD format a most clean one is use URL, but widely used is also property `X-SOCIALPROFILE` to VCard allow add different social networks like Twitter, Facebook and others:

```text
X-SOCIALPROFILE;type=twitter;x-user=twitteruser:http://twitter.com/twitteruser
X-SOCIALPROFILE;type=facebook;x-user=facebookuser:http://www.facebook.com/facebookuser
X-SOCIALPROFILE;type=flickr;x-user=flickruser:http://www.flickr.com/photos/flickruser
X-SOCIALPROFILE;type=linkedin;x-user=linkedinuser:http://www.linkedin.com/in/linkedinuser
X-SOCIALPROFILE;type=myspace;x-user=myspaceuser:http://www.myspace.com/myspaceuser
X-SOCIALPROFILE;type=sinaweibo;x-user=sinauser:http://weibo.com/n/sinauser
X-SOCIALPROFILE;type=Customsocial;x-user=custom socialuser:x-apple:custom%20socialuser
```

### Change description

Add method `public addSocial(url: string, type: string, user: string = '')` to the VCard object and update readme and tests.

### Check lists

- [x] Tests passed
- [x] Coding style respected

### CODE REVIEW

**Review by joaocarmo — CHANGES_REQUESTED** [review]
Thank you for the idea! Looks good overall, but I think that with a few small changes we can improve it.

Additionally, can we add some unit tests for it as well? Check the [VCard.test.ts](https://github.com/joaocarmo/vcard-creator/blob/9f8e3440897a4fd961ccc63c43cd341e8dec3645/lib/VCard.test.ts) file.

File: lib/VCard.ts [hunk 1]
```diff
@@ -407,6 +407,24 @@ ${lastName};${firstName};${additional};${prefix};${suffix}\
     return this
   }
 
+  /**
+   * Add social profile
+   *
+   * @param {string} url
+   * @param {string} type
+   * @param {string} user
```
[the remark this answers is not in the document]
Comment (OzzyCzech) [reply lib/VCard.ts]: this suggestions was add to #39

File: lib/types/VCard.ts [hunk 2]
```diff
@@ -18,6 +23,7 @@ export type Element =
   | 'role'
   | 'uid'
   | 'url'
+  | 'social'
```
[the remark this answers is not in the document]
Comment (OzzyCzech) [reply lib/types/VCard.ts]: this suggestions was add to #39

File: lib/VCard.ts [hunk 3]
```diff
@@ -407,6 +407,24 @@ ${lastName};${firstName};${additional};${prefix};${suffix}\
...
+   */
+  public addSocial(url: string, type: string, user: string = ''): this {
+    this.setProperty(
+      'social',
+      `X-SOCIALPROFILE${type !== '' ? `;type=${type}` : ''}${
+        user !== '' ? `;x-user=${user}` : ''
```
[the remark this answers is not in the document]
Comment (OzzyCzech) [reply lib/VCard.ts]: this suggestions was add to #39

**Pull request closed without merging**